simpleregex/tests/test_email.py

```diff
@@ -0,0 +1,319 @@
+                    "/",
+                    "=",
+                    "?",
+                    "^",
+                    "`",
+                    "{",
+                    "|",
+                    "}",
+                    "~",
+                    "-",
+                ]
+            )
+        )
+
+    @property
+    def local_alphanumeric(self):
+        """
+        Local alphanumeric part regex.
+        """
+        return none_or_many(
+            self.allowed_chars
+            + group(RegEx("\.") + self.allowed_chars, non_capturing=True)
+        )
+
+    @property
+    def domain_alphanumeric(self):
+        """
+        Alphanumeric regex for the domain part.
+        """
+        lower_alphanumeric = any_of_characters(
+            [regex_range("a", "z"), regex_range("0", "9")]
+        )
+        lower_alphanumeric_dash = any_of_characters(
+            [regex_range("a", "z"), regex_range("0", "9"), "-"]
+        )
+        return (
+            one_or_more(
+                group(
+                    lower_alphanumeric
+                    + none_or_one(
+                        group(
+                            none_or_many(lower_alphanumeric_dash)
+                            + lower_alphanumeric,
+                            non_capturing=True,
+                        )
+                    )
+                    + RegEx("\\."),
+                    non_capturing=True,
+                )
+            )
+            + lower_alphanumeric
+            + none_or_one(
+                group(
+                    none_or_many(lower_alphanumeric_dash) + lower_alphanumeric,
+                    non_capturing=True,
+                )
+            )
+        )
+
+    @property
+    def local_allowed_special_chars(self):
+        """
+        Special ascii characters allowed in local part.
+        """
+        return any_of_characters(
+            [
+                regex_range(r"\x01", r"\x08"),
+                r"\x0b",
+                r"\x0c",
+                regex_range(r"\x0e", r"\x1f"),
+                r"\x21",
+                regex_range(r"\x23", r"\x5b"),
+                regex_range(r"\x5d", r"\x7f"),
+            ]
+        )
+
+    @property
+    def domain_allowed_special_chars(self):
+        """
+        Special ascii characters allowed in domain part.
+        """
+        return any_of_characters(
+            [
+                regex_range(r"\x01", r"\x08"),
+                r"\x0b",
+                r"\x0c",
+                regex_range(r"\x0e", r"\x1f"),
+                regex_range(r"\x21", r"\x5a"),
+                regex_range(r"\x53", r"\x7f"),
+            ]
+        )
+
+    @property
+    def chars_allowed_after_backslash(self):
+        """
+        Special ascii characters allowed in local and domain part,
+        only if prefixed with a backslash.
+        """
+        return RegEx("\\") + any_of_characters(
+            [
+                regex_range(r"\x01", r"\x09"),
```

Comment:
I would change those chars into consts. Right now I don't know what I'm reading.

Reply:
it's just special ascii chars, it's not related to the package so no need to define them.